### ISSUE TITLE
OSC pins wrong certificate causing invalid certificate errors

### ISSUE THREAD

**Issue by Firstyear** [openSUSE/osc#1158]
```
WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
offending certificate is at '/home/886b45ce-98b2-4d79-9069-c9ff25bc0232/.config/osc/trusted-certs/api.suse.de_443.pem'
Certificate Verification Error: remote host identification has changed
```

This happens even 90~ish days, on whatever the api.suse.de refresh time is. This appears to be because OSC is incorrectly trusting the *leaf* certificate rather than the *intermediate*, so when the certificate is rotated it causes this validation error. 

This is bad because it trains our users that certificate errors are to be ignored, which allows trivial MITM attacks to be carried out. This could be considered a security issue due to this. 

@mgerstner @msmeissn

**Comment by msmeissn** [comment]
hmm, most people just import the SUSE SSL root CA ( from ca.suse.de ) instead of pinning.

But osc could pin both the intermediate and the leaf perhaps.

**Comment by dmach** [comment]
@Firstyear which osc version are you using? 0.x or 1.x?

**Comment by dmach** [comment]
I have discussed this with OBS people and we concluded that we don't want to support any kind if intermediate certificate pinning.
Self-signed certs can be explicitly trusted by the user, but everything else is supposed to be solved by installing trusted cacerts on the system (e.g. by installing `ca-certificates` or `ca-certificates-suse` package).

Please note that the latest osc-1.x doesn't allow trusting such certs and supports trusting self-signed certs only.

**Comment by Firstyear** [comment]
Maybe osc should ... depend on ca-certificates-suse then. That way we don't have this janky situation occuring.

**Comment by Firstyear** [comment]
@dmach the ca-certificates-suse package doesn't exist:


```

S  | Name                             | Summary                                               | Type
---+----------------------------------+-------------------------------------------------------+-----------
   | ca-certificates                  | Utilities for system wide CA certificate installation | srcpackage
i+ | ca-certificates                  | Utilities for system wide CA certificate installation | package
   | ca-certificates-cacert           | CAcert root certificates                              | srcpackage
   | ca-certificates-cacert           | CAcert root certificates                              | package
   | ca-certificates-mozilla          | CA certificates for OpenSSL                           | srcpackage
i+ | ca-certificates-mozilla          | CA certificates for OpenSSL                           | package
   | ca-certificates-mozilla-prebuilt | Pre-built CA certificates for OpenSSL                 | srcpackage
   | ca-certificates-mozilla-prebuilt | Pre-built CA certificates for OpenSSL                 | package
   | ca-certificates-steamtricks      | Provides /etc/ssl/certs/ca-certificates.crt           | srcpackage
   | ca-certificates-steamtricks      | Provides /etc/ssl/certs/ca-certificates.crt           | package
```

As a result, this issue remains valid. Can it be made available on tumbleweed, or some other fix made? Something else in the ca chain went to shit and the only way I can access api.suse.de atm is to set sslcheck to false.

**Comment by dmach** [comment]
> @dmach the ca-certificates-suse package doesn't exist:

It does, but it is only provided in an internal repo: https://download.suse.de/ibs/SUSE:/CA/
It would be convenient to have ca-certificates-suse in openSUSE repos, but why to pollute public repos with a package that nobody but SUSE employees needs?

**Comment by Firstyear** [comment]
Because a secure system can only exist if it is usable and simple. It's not "polluting" either.

Today, because of this certificate pinning issue, our employees are already being presented with a a request to change/trust the certificate and we are *training* them to always click trust. There are extremely good reasons why browsers started to try to avoid people clicking "trust" on certificates because most people even technically advanced ones are not qualified to correctly inspect a certificate for authenticity. 

Additionally it seems that on the latest TW changes to urllib/openssl now break the auto-trust prompt mechanism.

```
HTTPSConnectionPool(host='api.suse.de', port=None): Max retries exceeded with url: /source/home:firstyear:branches:SUSE:SLE-15-SP3:Update/_meta (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))
```

This error is at line 301 of osc/connection.py, meaning that the only way to currently bypass this and access IBS is either to install the certificates from that package, or to set sslcheck=off. Again, I had to set this last week to actually do my job because I did not have *time* to investigate a TLS issue. 

But if ca-certificates-suse was available by default then it would resolve all of these issues.

* A user will never have to examine or trust certificates for authenticity
* A user will not be trained to "trust" certificates every few days 
* Changes to SSL/TLS libraries will not create a scenario where a user reaches for sslcheck=0

Generally for us to improve the security of this we *must* make the defaults correct and valid. A person should not have to think about TLS and certificate management. 

Thus the only correct and secure option is to make ca-certificates-suse available in OBS, and to make it a dependency of osc so that certificate management is no longer an end user problem, and becomes a default secure system.
